### PR TITLE
20492 fdc3 combined searching

### PR DESCRIPTION
**Pull request by jason-ciq**
fix: #id [20492]

[Kanban link](https://chartiq.kanbanize.com/ctrl_board/18/cards/20492/details)

**Tag search now queries FDC3 server**
* Tag search used to just filter locally, breaking combinatory searches. Tags will now be sent to the appd server for searching.
* SearchText was moved to the distributed store. We need to know this in several places.
* goHome() was placed into store actions to properly react to combinatory searches.

**Description of testing**

1. Build and set up app catalog
1. Perform searches using tags and text. 
1. [x] Combinatory searches work properly, and when both tags and search text are cleared, app catalog returns to home.

### CODE REVIEW

File: src-built-in/components/advancedAppCatalog/src/stores/storeActions.js [hunk 1]
```diff
@@ -74,36 +85,24 @@ function getToolbarStore(done) {
 	});
 }
 
+function goHome() {
```
Comment (Garbee) [inline src-built-in/components/advancedAppCatalog/src/stores/storeActions.js]: Please add a jsdoc for this function.

File: src-built-in/components/advancedAppCatalog/src/stores/storeActions.js [hunk 2]
```diff
@@ -112,60 +111,38 @@ function _addActiveTag(tag) {
...
+
+function filterApps() {
+	let { activeTags, searchText } = data;
+	
+
+	if (activeTags.length === 0 && searchText === "") {
+		goHome();
+	} else {
+		appd.search({ text: searchText, tags: activeTags }, (err, data) => {
+			if (err) console.log("Failed to search apps");
```
Comment (Garbee) [inline src-built-in/components/advancedAppCatalog/src/stores/storeActions.js]: So, there was an error but we're going to continue to work with `data` that could be unusable information?

I'd expect a different exit pathway for error handling compared to continuing to work. If we can continue operations just fine, why log this at all?

Even if we do need this log, it should probably fall towards a warning. And also be logged to the Central Logger and not the DevTools console.

File: src-built-in/components/advancedAppCatalog/src/stores/storeActions.js [hunk 3]
```diff
@@ -112,60 +111,38 @@ function _addActiveTag(tag) {
...
 function _clearActiveTags() {
-	getStore().setValue({
-		field: "activeTags",
-		value: []
-	});
+	getStore().setValue({ field: "activeTags", value: [] });	
+}
+
+
+function filterApps() {
```
Comment (Garbee) [inline src-built-in/components/advancedAppCatalog/src/stores/storeActions.js]: JSDoc please.

File: src-built-in/components/advancedAppCatalog/src/stores/storeActions.js [hunk 4]
```diff
@@ -112,60 +111,38 @@ function _addActiveTag(tag) {
...
-		value: newActiveTags
-	}, {
-		field: "filteredApps",
-		value: newApps
-	}]);
+	getStore().setValue({ field: "activeTags", value: data.activeTags	});
+	filterApps();
 }
 
 function _refreshTags() {
```
Comment (Garbee) [inline src-built-in/components/advancedAppCatalog/src/stores/storeActions.js]: If all this is now doing is calling `filterApps` why don't we just update all calls to this to go straight to that and delete this?

File: src-built-in/components/advancedAppCatalog/src/stores/storeActions.js [hunk 5]
```diff
@@ -383,6 +360,27 @@ function clearFilteredApps() {
 	});
 }
 
+function setSearchValue(val) {
```
Comment (Garbee) [inline src-built-in/components/advancedAppCatalog/src/stores/storeActions.js]: JSDoc please.

File: src-built-in/components/advancedAppCatalog/src/stores/storeActions.js [hunk 6]
```diff
@@ -383,6 +360,27 @@ function clearFilteredApps() {
...
+	getStore().setValue({field: "searchText", value: val})
+}
+
+function getSearchValue() {
```
Comment (Garbee) [inline src-built-in/components/advancedAppCatalog/src/stores/storeActions.js]: JSDoc please.

File: src-built-in/components/advancedAppCatalog/src/stores/storeActions.js [hunk 7]
```diff
@@ -383,6 +360,27 @@ function clearFilteredApps() {
...
+	return data.searchText;
+}
+
+function clearSearchText() {
```
Comment (Garbee) [inline src-built-in/components/advancedAppCatalog/src/stores/storeActions.js]: JSDoc please.

File: src-built-in/components/advancedAppCatalog/src/stores/storeActions.js [hunk 8]
```diff
@@ -383,6 +360,27 @@ function clearFilteredApps() {
...
+	getStore().setValue({field: "searchText", value: ""})
+}
+
+function getForceSearch() {
```
Comment (Garbee) [inline src-built-in/components/advancedAppCatalog/src/stores/storeActions.js]: JSDoc please.

File: src-built-in/components/advancedAppCatalog/src/stores/storeActions.js [hunk 9]
```diff
@@ -383,6 +360,27 @@ function clearFilteredApps() {
...
+	return data.forceSearch;
+}
+
+
+function setForceSearch(val) {
```
Comment (Garbee) [inline src-built-in/components/advancedAppCatalog/src/stores/storeActions.js]: JSDoc please.

File: src-built-in/components/advancedAppCatalog/src/stores/storeActions.js (hunk 4 above)
Comment (jason-ciq) [reply src-built-in/components/advancedAppCatalog/src/stores/storeActions.js]: Reasonable. I'll take it out.

File: src-built-in/components/advancedAppCatalog/src/stores/storeActions.js (hunk 2 above)
Comment (jason-ciq) [reply src-built-in/components/advancedAppCatalog/src/stores/storeActions.js]: This was how it was previously written. However, I agree. Will log to central logger as an error and return.

File: src-built-in/components/advancedAppCatalog/src/stores/storeActions.js [hunk 10]
```diff
@@ -74,98 +85,67 @@ function getToolbarStore(done) {
...
+function filterApps() {
+	let { activeTags, searchText } = data;
+	
+
+	if (activeTags.length === 0 && searchText === "") {
+		goHome();
+	} else {
+		appd.search({ text: searchText, tags: activeTags }, (err, data) => {
+			if (err) {
+				Logger.system.error("FDC3 App search failed!: ${err}");
```
Comment (Garbee) [inline src-built-in/components/advancedAppCatalog/src/stores/storeActions.js]: (Clippy Voice): It looks like you're trying to interpolate a variable into a string. In JavaScript this should be done with back-ticks and not double quotes.

Comment (jason-ciq) [reply src-built-in/components/advancedAppCatalog/src/stores/storeActions.js]: Oops! Good catch. This has been resolved.

**Pull request merged**